pkcs12.go

```diff
@@ -32,11 +32,99 @@ import (
 )
 
 // DefaultPassword is the string "changeit", a commonly-used password for
-// PKCS#12 files. Due to the weak encryption used by PKCS#12, it is
-// RECOMMENDED that you use DefaultPassword when encoding PKCS#12 files,
-// and protect the PKCS#12 files using other means.
+// PKCS#12 files.
 const DefaultPassword = "changeit"
 
+// An Encoder contains methods for encoding PKCS#12 files.  This package
+// defines several different Encoders with different parameters.
+type Encoder struct {
+	macAlgorithm         asn1.ObjectIdentifier
+	certAlgorithm        asn1.ObjectIdentifier
+	keyAlgorithm         asn1.ObjectIdentifier
+	macIterations        int
+	encryptionIterations int
+}
```

Comment:
I would define these as public fields so a user can adjust the parameters. Some validation might be required.

Reply:
Or add a function to change the iterations, the rest of the parameters are not required in most changes. Something like this would be ideal so the user doesn't change the defined variables:

```go
func (e *Encoder) Iterations(iter int) (*Encoder) {
   return &Encoder {
      macAlgorith:          e.macAlgorithm,
      certAlgorithm:        e.certAlgorithm,
      keyAlgorithm:         e.keyAlgorithm,
      macIterations:        iter,
      encryptionIterations: iter,
   }
}
```

Reply:
Thanks for taking a look at this @maraino ! The `Iterations` method is a great idea; I've gone ahead and added `WithIterations`.  For now, I'm going to hold off on making the other parameters customizable.